shipitscript/task.py

```diff
@@ -28,29 +29,38 @@ def validate_task_schema(context):
     scriptworker.client.validate_json_schema(context.task, task_schema)
 
 
-def validate_task_scope(context):
+def get_ship_it_instance_config_from_scope(context):
     scope = _get_scope(context.task)
     allowed_api_root = ALLOWED_API_ROOT_PER_VALID_SCOPE[scope]
-    api_root_of_this_instance = context.config['ship_it_instance']['api_root']
-    api_root_of_this_instance = api_root_of_this_instance.rstrip('/')
-    if allowed_api_root != api_root_of_this_instance:
-        if allowed_api_root in PROTECTED_API_ROOTS or api_root_of_this_instance in PROTECTED_API_ROOTS:
-            instance_type = scope.split(':')[-1]
-            raise TaskVerificationError(
-                'This instance is now allowed to talk to the "{}" instance ({})'.format(instance_type, allowed_api_root)
-            )
+    configured_instances = context.config['ship_it_instances']
+    instance_type = scope.split(':')[-1]
 
+    if allowed_api_root in PROTECTED_API_ROOTS:
```

Comment:
This works, but I think we can get rid of this check if the instance info is in config.
e.g., dep signing scriptworkers just don't have any pw info for the nightly/release signing servers.
[dep pw](https://hg.mozilla.org/build/puppet/file/tip/modules/signing_scriptworker/templates/dep-passwords.json.erb)
[non-dep pw](https://hg.mozilla.org/build/puppet/file/tip/modules/signing_scriptworker/templates/passwords.json.erb)

Reply:
Yeah, that's true. Moreover, this simplifies the code. I followed up on it in #3